Add mcxc444 pwm support

NeilChen93 opened the pull request:
Add mcxc444 pwm support

NeilChen93 committed:
boards: nxp: frdm_mcxc444: Update led pin definitions

Fixing the DT LED definitions

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

simonguinot commented on boards/nxp/frdm_mcxc444/frdm_mcxc444.dts:
```diff
@@ -36,7 +36,7 @@
 		};
 		green_led: led_1 {
-			gpios = <&gpiob 5 GPIO_ACTIVE_LOW>;
+			gpios = <&gpiod 5 GPIO_ACTIVE_LOW>;
```
It looks more like you are "fixing the DT LED definitions" than "Update led pin align with board SCH" :) You should just say it in the commit title and message.

NeilChen93 committed:
boards: frdm_mcxc444: Add pwm support

Support pwm for NXP frdm_mcxc444 board.
Test it using samples/basic/blinky_pwm.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

simonguinot commented on samples/basic/blinky_pwm/boards/frdm_mcxc444.overlay:
```diff
@@ -0,0 +1,9 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Is that OK to keep the `gpio-leds` node enabled here ?

simonguinot commented on samples/basic/blinky_pwm/boards/frdm_mcxc444.overlay:
```diff
@@ -0,0 +1,9 @@
+ */
+
+&tpm0 {
+	clocks = <&sim KINETIS_SIM_OSCERCLK 0x103C 24>;
```
Shouldn't it be in the board DTS file ?

NeilChen93 committed:
samples: pwm: Add overlay for frdm_mcxc444 board

Add Devicetree overlay for samples drivers/led/pwm to enable pwmleds.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>
